lib/cppapi/GzipDeflateTransformation.cc

```diff
@@ -94,7 +93,7 @@ GzipDeflateTransformation::consume(const string &data)
 
   int iteration               = 0;
   state_->z_stream_.data_type = Z_ASCII;
-  state_->z_stream_.next_in   = reinterpret_cast<unsigned char *>(const_cast<char *>(data.c_str()));
+  state_->z_stream_.next_in   = reinterpret_cast<unsigned char *>(const_cast<char *>(data.data()));
```

Comment:
`data()` isn't guaranteed to return a nul-terminated buffer.

Reply:
Looking at the zlib documentation, I can't see any requirement that the sequence of bytes pointed to by next_in be nul terminated.

Reply:
Ah I see, there's a `avail_in` member being set too. Makes sense.